api/routes/utils.js

```diff
@@ -642,228 +577,68 @@ var utils = {
   },
 
   // Given a transaction, update all dividends for investors investing in that expert
-  updateDividends: function(investors, categoryName, expertName, expertReps) {
-    var investor, j, errs;
-    var length = investors.length;
-    for (var i = 0; i < length; i++) {
-      j = -1;
-      investor = investors[i];
-      j = this.getPortfolioIndex(investor, categoryName);
-      if (j === -1) {
-        continue;
-      }
-
-      // Find the corresponding investment and reset the dividend
-      var len = investor.portfolio[j].investments.length;
-      for (var q = 0; q < len; q++) {
-        var investment = investor.portfolio[j].investments[q];
-        if (investment.user === expertName) {
-          investment.dividend = Math.round(investment.percentage * expertReps * DIVIDEND_RATE * 100) / 100;
+  updateDividends: function(expert, categoryName, cb) {
+    var expertReps;
+    var i = this.getCategoryIndex(expert, categoryName);
+    if (i < 0) {
+      return cb('User is not an expert in this category');
+    }
+    expertReps = expert.categories[i].reps;
+
+    User.findInvestments(expert._id, categoryName).then(function(investmentList) {
+      var investments, investment;
+      for (var i = 0; i < investmentList.length; i++) {
+        investments = investmentList[i].investments;
+        // Find the corresponding investment and reset the dividend
+        for (var j = 0; j < investments.length; j++) {
+          investment = investments[j];
+          if (investment.userId.toString() === expert._id.toString()) {
+            investment.dividend = Math.round(investment.percentage * expertReps * DIVIDEND_RATE * 100) / 100;
+          }
         }
+        // Update the current user with the new dividend
+        User.updateInvestments(investmentList[i]._id, categoryName, investments);
```

Comment:
Notice, we don't wait on the result of this action, we just continue. We wouldn't stop if there were an issue, and the mongo middleware isn't even active here